pkg/registry/generic/rest/proxy.go

```diff
@@ -174,23 +174,53 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(h.Location)
 
+	var dialer func(network, addr string) (net.Conn, error)
```

Comment:
Can you add a 

// TODO: this TLS logic can probably be cleaned up; it's messy in an attempt to preserve behavior that we don't know for sure is exercised.

And then I will stop bothering you about it :)